info-compiler/src/main/java/com/google/sps/data/NewsArticle.java

```diff
@@ -51,4 +58,27 @@ public void setContent(String content) {
   public void setAbbreviatedContent(String abbreviatedContent) {
     this.abbreviatedContent = abbreviatedContent;
   }
+
+  /**
+   * Compares whether individual instance variables are equal, with the exception that content
+   * can be the substring of another without being exactly the same. The exception is allowed
+   * because, when scraping real webpages, the scraped content also includes the actual news
+   * articles body as well as changing content such as ads, descriptions of other articles.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof NewsArticle)) {
+      return false;
+    }
+    NewsArticle other = (NewsArticle) obj;
+    if (other == this) {
```

Comment:
I would do the object comparison first and then check the instanceof. If obj is already == this, you're wasting a step by casting it to NewsArticle.